platform/ios/src/MGLMapView.mm

```diff
@@ -4231,6 +4361,84 @@ - (void)updateUserLocationAnnotationView
     [self updateUserLocationAnnotationViewAnimatedWithDuration:0];
 }
 
+- (void)updateAnnotationViews
+{
+    BOOL delegateImplementsViewForAnnotation = [self.delegate respondsToSelector:@selector(mapView:viewForAnnotation:)];
+    
+    if (!delegateImplementsViewForAnnotation)
+    {
+        return;
+    }
+    
+    // Update all visible annotation views
+    std::set<MGLAnnotationTag> visibleTags;
+    std::vector<MGLAnnotationTag> annotationTags = [self annotationTagsInRect:CGRectInset(self.bounds, -_largestAnnoationViewWidth, -_largestAnnotationViewHeight)];
+    
+    for(auto const& annotationTag: annotationTags)
+    {
+        auto &annotationContext = _annotationContextsByAnnotationTag[annotationTag];
+        id<MGLAnnotation> annotation = annotationContext.annotation;
+        
+        CGPoint annotationPoint = [self convertCoordinate:annotation.coordinate toPointToView:self];
+       
+        // If there is no annotation view at this point, it means the context's view was reused by some
+        // other context so we need to reuse or make a new view.
+        if (!annotationContext.annotationView)
+        {
+            MGLAnnotationView *annotationView = [self annotationViewForAnnotation:annotation];
+            
+            if (annotationView)
+            {
+                // If the annotation view has no superview it means it was never used before so add it
+                if (!annotationView.superview)
+                {
+                    [self.glView addSubview:annotationView];
+                }
+                
+                annotationContext.annotationView = annotationView;
+            }
+        }
+       
+        annotationContext.annotationView.center = annotationPoint;
+        annotationContext.annotationView.hidden = annotationContext.annotationView.hidden ?: NO;
```

Comment:
oops, I did not mean to commit this. I was experimenting with ways to maintain the hidden state of a view for an application developer but I eventually came to the conclusion that it is best to let the developer have complete control over hidden while the view is visible. When it goes off screen we enqueue / hide it. When it is dequeued from the reuse queue we unhide it (an application developer can re-hide it if required when they get it back from the reuse queue). Tail work might be to investigate why `annotationTagsInRect` (and the underlying boost query) does not return some tags that are definitely visible when the map is tilted  (which is why we are currently hiding as a workaround)